client/ctclient/ctclient.go

```diff
@@ -216,6 +219,63 @@ func getInclusionProofForHash(ctx context.Context, logClient *client.LogClient,
 	}
 }
 
+func getConsistencyProof(ctx context.Context, logClient *client.LogClient) {
+	if *treeSize <= 0 {
+		log.Fatal("No valid --size supplied")
+	}
+	if *prevTreeSize <= 0 {
+		log.Fatal("No valid --prev_size supplied")
+	}
+	var hash1, hash2 []byte
+	if *prevHash != "" {
+		var err error
+		hash1, err = hex.DecodeString(*prevHash)
+		if len(hash1) != 32 {
+			err = fmt.Errorf("invalid hash length %d", len(hash1))
+		}
+		if err != nil {
+			log.Fatalf("Invalid --prev_hash: %v", err)
+		}
+	}
+	if *treeHash != "" {
+		var err error
+		hash2, err = hex.DecodeString(*treeHash)
+		if len(hash1) != 32 {
+			err = fmt.Errorf("invalid hash length %d", len(hash2))
+		}
+		if err != nil {
+			log.Fatalf("Invalid --tree_hash: %v", err)
+		}
+	}
+	getConsistencyProofBetween(ctx, logClient, *prevTreeSize, *treeSize, hash1, hash2)
+}
+
+func getConsistencyProofBetween(ctx context.Context, logClient *client.LogClient, first, second int64, prevHash, treeHash []byte) {
+	proof, err := logClient.GetSTHConsistency(ctx, uint64(first), uint64(second))
+	if err != nil {
+		if err, ok := err.(client.RspError); ok {
```

Comment:
As an aside, I wonder whether `RspError` should just be including all the info in the string it returns from `Error()` - it seems quite unwieldy to make every calling site try to figure out what type of error was returned and handle them differently!